### PR TITLE
fix bug in loo and p_loo computation, also return waic and loo result as dataframes

### PR DESCRIPTION
**Pull request by aloctavodia**
While trying to address what we discuss in #1677 I found that a previous PR introduced an error in the computation of `p_loo` (a variable was undefined). This PR fix it. I also found a bug in the computation of lppd_loo related to the _unsorting_ of the importance ratios. Now the results of  `pm.loo` are closer to those reported for the 8 school problem in the papers "Understanding predictive information criteria for Bayesian models" and "Practical Bayesian model evaluation using leave-one-out cross-validation and WAIC".

### CODE REVIEW

File: pymc3/stats.py [hunk 1]
```diff
@@ -151,16 +150,18 @@ def waic(trace, model=None, n_eff=False, pointwise=False):
 
     p_waic = np.sum(vars_lpd)
 
-
-    if n_eff:
-        return waic, waic_se, p_waic
-    elif pointwise:
-        return waic, waic_se, waic_i, p_waic
+    if pointwise:
+        return pd.DataFrame([[waic, waic_se, p_waic, waic_i]],
```
Comment (twiecki) [inline pymc3/stats.py]: This should be a pd.Series I think (or maybe a namedtuple).

File: pymc3/stats.py [hunk 2]
```diff
@@ -220,24 +224,30 @@ def loo(trace, model=None, n_eff=False):
...
-    r_new = np.array([r[np.argsort(i)]
-                      for r, i in zip(r_sorted, np.argsort(r, axis=0))])
+    r_new =  np.array([r[np.argsort(i)]
+                      for r, i in zip(r_sorted.T, np.argsort(r.T, axis=1))]).T
 
     # Truncate weights to guarantee finite variance
     w = np.minimum(r_new, r_new.mean(axis=0) * S**0.75)
 
-    loo_lppd_i = -2.0 * logsumexp(log_py, axis = 0, b = w / np.sum(w, axis = 0))
+    loo_lppd_i = - 2. * logsumexp(log_py, axis=0, b=w/np.sum(w, axis=0))
```
Comment (twiecki) [inline pymc3/stats.py]: `b=w / np.sum...`

Comment (twiecki) [reply pymc3/stats.py]: Spaces around math operators.

File: pymc3/stats.py (hunk 1 above)
Comment (twiecki) [reply pymc3/stats.py]: Yeah, I think namedtuple would be best here as we also use it for e.g. ADVI.

File: pymc3/stats.py (hunk 2 above)
Comment (aloctavodia) [reply pymc3/stats.py]: ups! I will fix it

File: pymc3/stats.py (hunk 1 above)
Comment (aloctavodia) [reply pymc3/stats.py]: I thought the same, but Jupyter renders DataFrame in a nicer way. And I am also thinking this is more consistent with a _compare_ function (that I need to write) that will display waic/loo results for several models as a DataFrame.

Comment (aloctavodia) [reply pymc3/stats.py]: @twiecki still thinking namedtuple is the way to go? If so let me now and I will change it.

Comment (twiecki) [reply pymc3/stats.py]: Your argument is purely for display purposes? I'm not convinced that should guide API and data-structure choices.

Comment (twiecki) [reply pymc3/stats.py]: It's possible to convert, but a bit clunky:
```python
from collections import namedtuple
Point = namedtuple('test', ['x', 'y'])
p = Point(1, 2)
pd.Series(p._asdict()).to_frame()
```

Comment (aloctavodia) [reply pymc3/stats.py]: yep purely aesthetics. OK. I will use namedtuples. Thanks for the review.

**Comment by twiecki** [comment]
Thanks @aloctavodia!

**Pull request merged**